common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -411,13 +411,17 @@ private void record0(Object hint) {
                 Record newHead;
                 boolean dropped;
                 do {
+                    dropped = false;
                     if ((prevHead = oldHead = headUpdater.get(this)) == null) {
                         // already closed.
                         return;
                     }
-                    int numElements = oldHead.pos + 1;
-                    if (dropped = PlatformDependent.threadLocalRandom().nextInt(1 << numElements) >= TARGET_RECORDS) {
-                        prevHead = oldHead.next;
+                    final int numElements = oldHead.pos + 1;
+                    if (numElements >= TARGET_RECORDS) {
+                        final int backOffFactor = Math.min(numElements - TARGET_RECORDS, 31);
```

Comment:
`1<<31 == -2147483648` which will throw an IllegalArgumentException. this should max out at `30` or you could consider using `(1 << backOffFactor) - 1` instead.

Reply:
I picked 30.  Done.